librad/src/net/protocol/membership/hpv.rs

```diff
@@ -315,7 +315,15 @@ where
     }
 
     pub fn choose_passive_to_promote(&mut self) -> Vec<PeerInfo<Addr>> {
-        let n = self.params.max_active - self.num_active();
+        assert!(
+            self.params.max_active >= self.num_active(),
+            "number of active peers is larger than the configured max"
+        );
+        let n = self
+            .params
+            .max_active
+            .checked_sub(self.num_active())
+            .unwrap_or(1);
```

Comment:
Belts and suspenders?

Reply:
Sicher ist sicher. Happy to revise this.